Sources/NIOHTTP2/HTTP2StreamChannel.swift

```diff
@@ -694,14 +694,14 @@ final class HTTP2StreamChannel: Channel, ChannelCore, @unchecked Sendable {
         self.failPendingWrites(error: error)
         if let promise = self.pendingClosePromise {
             self.pendingClosePromise = nil
-            promise.fail(error)
+            promise.succeed()
```

Comment:
I don't think this is what was agreed: IIRC the promise from `close(mode:promise:)` (i.e. `pendingClosePromise`)  _can_ fail to provide diagnostic information. It's the `closeFuture` (i.e. part of the `Channel` API) which mustn't fail.

Reply:
Agree with what @glbrntt said here. This means we need to change the implementation of `executeThenClose` to discard the error from `close` and instead add a call to `closeFuture.get()`

Reply:
And update [the docs](https://github.com/apple/swift-nio/blob/ecfaa2c6ff3234bdf7909eba0da01c245922e471/Sources/NIOCore/Channel.swift#L109) for `Channel.closeFuture`

Reply:
Oh okay, I misunderstood this. The one thing I found weird when implementing this (and why I decided to not fail this promise either) is that calling `channel.close()` will return a failed promise (since it creates one and passes it to `channel.close(promise: newPromise)` under the hood), which I personally find slightly confusing. But if you're okay with this behaviour I can just be clear in the docs about it.

Reply:
That's expected. Roughly speaking:
- `channel.close(promise:)` (and variants of, i.e. `channel.close()`) are "close the channel and tell me _how_ it closed" (cleanly/uncleanly) 
- `channel.closeFuture` is "tell me _when_ the channel is closed"